server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -83,9 +83,24 @@ public class RemoteClusterStateService implements Closeable {
 
     private static final Logger logger = LogManager.getLogger(RemoteClusterStateService.class);
 
-    // TODO make this two variable as dynamic setting [issue: #10688]
-    public static final int INDEX_METADATA_UPLOAD_WAIT_MILLIS = 20000;
-    public static final int GLOBAL_METADATA_UPLOAD_WAIT_MILLIS = 20000;
+    // default value for index metadata upload wait is 20s
+    static volatile TimeValue indexMetadataUploadWaitTime = TimeValue.timeValueMillis(20000);
+    // default value for global metadata upload wait is 20s
+    static volatile TimeValue globalMetadataUploadWaitTime = TimeValue.timeValueMillis(20000);
```

Comment:
Keep the constants naming convention, uppercase separated by _

Reply:
Makes sense, updated this